app/models/dialog/terraform_template_service_dialog.rb

```diff
@@ -0,0 +1,92 @@
+require "JSON"
+
+class Dialog
+  class TerraformTemplateServiceDialog
+    def self.create_dialog(label, terraform_template, extra_vars)
+      new.create_dialog(label, terraform_template, extra_vars)
+    end
+
+    # This dialog is to be used by a terraform template service item
+    def create_dialog(label, terraform_template, extra_vars)
+      Dialog.new(:label => label, :buttons => "submit,cancel").tap do |dialog|
+        tab = dialog.dialog_tabs.build(:display => "edit", :label => "Basic Information", :position => 0)
+        if terraform_template.present?
+          add_template_variables_group(tab, 0, terraform_template)
+        end
+        if extra_vars.present?
+          add_variables_group(tab, 1, extra_vars)
+        end
+        dialog.save!
+      end
+    end
+
+    private
+
+    def add_template_variables_group(tab, position, terraform_template)
+      template_info = JSON.parse(terraform_template.payload)
+      input_vars = template_info["input_vars"]
+
+      return if input_vars.nil?
+
+      tab.dialog_groups.build(
+        :display  => "edit",
+        :label    => "Terraform Template Variables",
+        :position => position
+      ).tap do |dialog_group|
+        input_vars.each_with_index do |(var_info), index|
+          key = var_info["name"]
+          value = var_info["default"]
+          value = value.to_json if [Hash, Array].include?(value.class)
+          required = var_info["required"]
+          readonly = var_info["immutable"]
+          hidden = var_info["hidden"]
+          label = var_info["label"]
+          description = var_info["description"]
+          if description.blank?
+            description = key
+          end
```

Comment:
This is a good time to one-line this
```suggestion
          description = key if description.blank?
```